src/equity/equity.controller.ts

```diff
@@ -0,0 +1,22 @@
+import { Controller, Get, UseGuards, Req } from '@nestjs/common';
+import { Request } from 'express';
+import { Local } from '~/decorator';
+import { JwtMember } from '~/auth/auth.dto';
+import { ProgramService } from '~/program/program.service';
+import { AuthGuard } from '~/auth/auth.guard';
+
+@UseGuards(AuthGuard)
+@Controller({
+  path: 'equity',
+  version: '2',
+})
+export class EquityController {
+  constructor(private programService: ProgramService) {}
+
+  @Get('/programs')
+  async getProgramsByMemberId(@Local('member') member: JwtMember, @Req() request: Request) {
```

Comment:
@Req decorator is too powerful , we can just use @Query('memberId') memberId: string to restrict the params
or add new dto class , like GetProgramByMemberIdsRequestDto to check params type valid 
```
class GetProgramByMemberIdsRequestDto {
  @IsString()
  memberId
 }
```

Reply:
it is better to use inline code block to present `@Req` instead of avoid tagging unknown person.

Reply:
FYI: https://zh.wikipedia.org/zh-tw/%E6%9C%80%E5%B0%8F%E6%9D%83%E9%99%90%E5%8E%9F%E5%88%99

Reply:
maybe there is a kind of feature like this:
```
Nest.query.addValidator('memberId', value => {
     return /^[0-9|a-z]{9}/.test(value)
})
@query('memberId') memberId
```
so this way cannot apply the feature
```
@req request
{memberId} = request.query
```